prelude.dx

```diff
@@ -801,24 +801,39 @@ String : Type = List Char
 
 CharPtr : Type = %CharPtr
 
-interface Show a:Type where
-  show : a -> String
-
 -- TODO. This is ASCII code point. It really should be Int32 for Unicode codepoint 
 def codepoint (c:Char) : Int8 = %codePoint c
 
 @instance charEq  : Eq Char  = MkEq \x y. codepoint x == codepoint y
 @instance charOrd : Ord Char = (MkOrd charEq (\x y. codepoint x > codepoint y)
                                              (\x y. codepoint x < codepoint y))
 
+interface Show a:Type where
+  show : a -> String
 
-def showFloat' (x:Float) : String =
-  (n, ptr) = %ffi showFloat (Int & CharPtr) x
-  AsList n $ for i:(Fin n).
-    MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
-
-instance showFloat : Show Float where
-  show = showFloat'
+instance showInt32 : Show Int32 where
+  show = \x: Int32.
+    (n, ptr) = %ffi showInt32 (Int & CharPtr) x
+    AsList n $ for i:(Fin n).
+      MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+instance showInt64 : Show Int64 where
+  show = \x: Int64.
+    (n, ptr) = %ffi showInt64 (Int & CharPtr) x
+    AsList n $ for i:(Fin n).
+      MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+instance showFloat32 : Show Float32 where
+  show = \x: Float32.
+    (n, ptr) = %ffi showFloat32 (Int & CharPtr) x
+    AsList n $ for i:(Fin n).
+      MkChar $ %ptrLoad (%ptrOffset ptr (ordinal i))
+
+instance showFloat64 : Show Float64 where
+  show = \x: Float64.
+    (n, ptr) = %ffi showFloat64 (Int & CharPtr) x
```

Comment:
Can you please use `Int32` instead of `Int`? We're thinking about making the precision of `Int` configurable, but this does not play well with FFI, which should use fixed-width types!

Reply:
Good point, done!